doc/api.md

```diff
@@ -124,6 +124,62 @@ Adds a row to the `mail_unsubscribes` table, preventing email form being sent to
 
 Requires an authentication token granting admin privileges. A request `Content-Type` of `application/x-www-form-urlencoded` is required. Renders the template with the data in the body.
 
+User Management
+---------------
+
+Most of these APIs operate on the data stored in members' records. The following mapping from database schema to JSON representation (as keys on a user object) is established for use with this API:
+ - `id`: The internal database ID is exposed as an integer `db_id`. This may not be null.
+ - `name`: The name is exposed as the string `name`. This may not be null.
+ - `x500`: The (psuedo-)x500 is exposed as the string `x500`. This may not be null.
+ - `card`: The long UCard number is exposed as the string `card`. This must be exactly 17 characters. This may be null instead.
+ - `email`: The email is exposed as the string `email`. This may not be null.
+ - `discordID`: The discord ID is exposed as the string `discordID`. This (as opposed to an integer) is adopted for the same (bad) reasons as the official Discord API, as well as interop with said API. This may be null.
+
+The term "canonical user object" is used to refer to a user object with all of the preceding keys set. User objects that are missing a key may be used to e.g. represent a change to make. Missing keys do not mean the value of the field is null.
```

Comment:
Ditto with missing key == null; I'd rather `PUT` semantics than `apply-diff` ones.

Reply:
I'm not wedded to this scheme but I think `apply-diff` is conceptually cleaner here - your'e not _replacing_ the user, your updating it in place. Plus, it prevents ABA.